ibm/service/power/resource_ibm_pi_network_security_group_rule.go

```diff
@@ -0,0 +1,654 @@
+							Optional:    true,
+							Type:        schema.TypeInt,
+						},
+					},
+				},
+				ForceNew: true,
+				MaxItems: 1,
+				Optional: true,
+				Type:     schema.TypeList,
+			},
+			Arg_NetworkSecurityGroupID: {
+				Description: "The unique identifier of the network security group.",
+				ForceNew:    true,
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_NetworkSecurityGroupRuleID: {
+				ConflictsWith: []string{Arg_Action, Arg_DestinationPorts, Arg_Protocol, Arg_Remote, Arg_SourcePorts},
+				Description:   "The network security group rule id to remove.",
+				ForceNew:      true,
+				Optional:      true,
+				Type:          schema.TypeString,
+			},
+			Arg_Protocol: {
+				ConflictsWith: []string{Arg_NetworkSecurityGroupRuleID},
+				Description:   "The protocol of the network traffic.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_ICMPTypes: {
+							Description: "If icmp type, the list of ICMP packet types (by numbers) affected by ICMP rules and if not present then all types are matched.",
+							Elem:        &schema.Schema{Type: schema.TypeInt},
+							Required:    true,
+							Type:        schema.TypeList,
+						},
+						Attr_TCPFlags: {
+							Description: "If tcp type, the list of TCP flags and if not present then all flags are matched.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									Attr_Flag: {
+										Description: "TCP flag.",
+										Required:    true,
+										Type:        schema.TypeString,
+									},
+								},
+							},
+							Optional: true,
+							Type:     schema.TypeList,
+						},
+						Attr_Type: {
+							Description:  "The protocol of the network traffic.",
+							Required:     true,
+							Type:         schema.TypeString,
+							ValidateFunc: validate.ValidateAllowedStringValues([]string{All, ICMP, TCP, UDP}),
+						},
+					},
+				},
+				ForceNew: true,
+				MaxItems: 1,
+				Optional: true,
+				Type:     schema.TypeList,
+			},
+			Arg_Remote: {
+				ConflictsWith: []string{Arg_NetworkSecurityGroupRuleID},
+				Description:   "The protocol of the network traffic.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_ID: {
+							Description: "The ID of the remote network address group or network security group the rules apply to. Not required for default-network-address-group.",
+							Optional:    true,
+							Type:        schema.TypeString,
+						},
+						Attr_Type: {
+							Description:  "The type of remote group (MAC addresses, IP addresses, CIDRs, external CIDRs) that are the originators of rule's network traffic to match.",
+							Optional:     true,
+							Type:         schema.TypeString,
+							ValidateFunc: validate.ValidateAllowedStringValues([]string{"default-network-address-group", "network-address-group", "network-security-group"}),
```

Comment:
Could you add them to the const file?
`DefaultNAG, NAG, NSG`

Reply:
updated